docker/release.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+# Usage:
+# // To build and publish the Docker image to the internal container registry.
+# $ DESTINATION=internal INTERNAL_CONTAINER_REGISTRY=<internal_registry> ./release.sh
+# // To publish the Docker image to the public container registry.
+# $ DESTINATION=public ./release.sh
+set -ex
+
+PUBLIC_CONTAINER_REGISTRY='gcr.io/stackdriver-agents/stackdriver-logging-agent'
+
+source VERSION
+# TODO(qingling128): Remove this suffix when kubernetes-branch is deprecated.
+IMAGE_SUFFIX=""
+DOCKER_IMAGE_VERSION="${DOCKERFILE_VERSION}-${GOOGLE_FLUENTD_VERSION}${IMAGE_SUFFIX}"
+DESTINATION=${DESTINATION:-internal}
+
+if [[ -z "${INTERNAL_CONTAINER_REGISTRY}" ]]; then
+  echo "Please provide a INTERNAL_CONTAINER_REGISTRY environment variable.";
+  exit 1;
+fi
+
+case "${DESTINATION}" in
+  internal)
+    echo "Building ${DOCKER_IMAGE_VERSION}.";
+    docker build --no-cache -t "${INTERNAL_CONTAINER_REGISTRY}:${DOCKER_IMAGE_VERSION}" .;
+    echo "Releasing ${DOCKER_IMAGE_VERSION} to the internal container registry.";
+    docker push "${INTERNAL_CONTAINER_REGISTRY}:${DOCKER_IMAGE_VERSION}";
```

Comment:
I think we should leave the image name as stable, and vary only the registry and the image version.

Right now, the `INTERNAL_CONTAINER_REGISTRY` could be set to something like `gcr.io/stackdriver-agents/stackdriver-logging-agent`, I believe the right thing  to do instead is set `INTERNAL_CONTAINER_REGISTRY` to `gcr.io/stackdriver-agents`, and then append `/stackdriver-logging-agent:${DOCKER_IMAGE_VERSION}`, or at least make `stackdriver-logging-agent` a `DOCKER_IMAGE_NAME` with a default. WDYT?

Reply:
To be totally transparent, I've gained some inspiration by how heapster handles this in their makefile, but that does not necessarily mean it's best practice or recommended.

https://github.com/kubernetes/heapster/blob/master/Makefile#L76

Reply:
Heapster has a public staging repo. We don't.